### PR TITLE
Create value class for layout modifier tag

### LINKED ISSUE #418: Make protocol values all value types
~There's no reason to use signed values.~ Just going to use value types instead. Unsigned values inside value types bad. See below..

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #418

### CODE REVIEW

File: redwood-protocol/src/commonMain/kotlin/app/cash/redwood/protocol/protocol.kt [hunk 1]
```diff
@@ -50,13 +57,37 @@ public data class PropertyDiff(
...
-  /**
-   * Array of layout modifiers. Each element of this array is itself a two-element array of the
-   * layout modifier tag and then serialized value.
-   */
-  val elements: JsonArray,
+  val elements: List<LayoutModifierElement> = emptyList(),
 )
 
+@Serializable(with = LayoutModifierElementSerializer::class)
+public data class LayoutModifierElement(
```
Comment (swankjesse) [inline redwood-protocol/src/commonMain/kotlin/app/cash/redwood/protocol/protocol.kt:64]: Yay, better structure.

File: redwood-protocol/src/commonMain/kotlin/app/cash/redwood/protocol/protocol.kt [hunk 2]
```diff
@@ -50,13 +57,37 @@ public data class PropertyDiff(
...
+
+private object LayoutModifierElementSerializer : KSerializer<LayoutModifierElement> {
+  private val delegate = serializer<Array<JsonElement>>()
+
+  @OptIn(ExperimentalSerializationApi::class)
+  override val descriptor = SerialDescriptor("LayoutModifierElement", delegate.descriptor)
+
+  override fun deserialize(decoder: Decoder): LayoutModifierElement {
+    val decoded = decoder.decodeSerializableValue(delegate)
+    check(decoded.size in 1..2) { "TODO" }
```
Comment (swankjesse) [inline redwood-protocol/src/commonMain/kotlin/app/cash/redwood/protocol/protocol.kt]: lenient?

```
check(decoded.size >= 2)
```

**Review by swankjesse — APPROVED** [review]
I like pushing the compact encoding work to the serializer

File: redwood-protocol/src/commonMain/kotlin/app/cash/redwood/protocol/protocol.kt (hunk 2 above)
Comment (JakeWharton) [reply redwood-protocol/src/commonMain/kotlin/app/cash/redwood/protocol/protocol.kt]: 0 elements are also invalid. The tag is required.

But I'll add the message I forgot and maybe write a test for this logic, too.

**Comment by JakeWharton** [comment]
Well I'm glad I wrote tests. This is trickier than it seems!

**Pull request merged**